src/main.rs

```diff
@@ -173,52 +168,54 @@ fn display_wallet_balance(wallet_file_name: &PathBuf, long_form: Option<bool>) {
     println!("coin count = {}", utxo_count);
     println!("total balance = {}", balance);
 
-    let secp = Secp256k1::new();
     let incomplete_coinswaps = wallet.find_incomplete_coinswaps(&rpc).unwrap();
     if incomplete_coinswaps.len() > 0 {
         println!("= incomplete coinswaps =");
-        for (hashvalue, utxo_swapcoins) in incomplete_coinswaps {
-            let balance: Amount = utxo_swapcoins
+        for (hashvalue, (utxo_incoming_swapcoins, utxo_outgoing_swapcoins)) in incomplete_coinswaps
+        {
+            let balance: Amount = utxo_incoming_swapcoins
                 .iter()
-                .fold(Amount::ZERO, |acc, us| acc + us.0.amount);
+                .zip(utxo_outgoing_swapcoins.iter())
```

Comment:
This assumes that the number of incoming swapcoins is the same as the number of outgoing swapcoins, which isn't always true (for example it wont be true if a coinswap has stalled, because then some peers will have outgoing swapcoins but no corresponding incoming swapcoin). I'd recommend using Iterator.chain() instead or something else similar.

Reply:
Fixed